### ISSUE TITLE
Feedparser - Fails to retrieve a date from episodes when the date format ends with a three letter timezone (e.g. PST)

### ISSUE THREAD

**Issue by mitchdowney** [podverse/podverse#49]
I noticed this issue when subscribing to the Dan Carlin Hardcore History podcast. All of the episodes are appearing in the EpisodeTableViewController in reverse chronological order. 

The episodes are failing to appear in chronological order in the EpisodeTableViewController because the FeedParser does not correctly grab the pubDate of the DCHH podcast. When the FeedParser fails to grab a pubDate, it defaults to setting the pubDate as the current date (for all the episodes).

So the format of the pubDate in the [DCHH RSS Feed](http://feeds.feedburner.com/dancarlin/history?format=xml) (view source of that page) looks like this:

<pubDate>Tue, 05 May 2015 23:57:12 PST</pubDate>

After investigating the NSDate+FeedsDate.swift file (which is responsible for formating the FeedParser's dates), I can see that when parsing the DCHH feed the NSDate+FeedsDate fails in line 52 to recognize DCHH's pubDate as a valid date format because the DCHH pubDate does not end with the letter "Z".

(Apparently three letter time zones like PST or MST or CST or EST are according to web standards supposed to have a Z at the end of their pubDate? DCHH doesn't have a Z, and it doesn't match the other expected date formats, so retrieveDateFromDateString in FeedParser.swift defaults to the current date.)

I dunno how to solve this one right now. It's very low priority so I'm moving on...

**Comment by mitchdowney** [comment]
@kreonjr I'd appreciate ur help with this one, might be easy but I'm not sure how to handle it.

**Comment by mitchdowney** [comment]
After reviewing this list of time zones, it looks like all non-military time zones end in either a T, K, D, C, B, or A.

It looks like the Feedparser is currently looking for a Z at the end of the time zone to signify the time format. Should we look for TKDCBA at the end too?

**Comment by mitchdowney** [comment]
Hmmm more time zone issues are happening with the [TED Radio Hour RSS feed](http://www.npr.org/rss/podcast.php?id=510298).

It seems the NSDate+FeedsDate.swift is setup only to handle +00:00 and +0000, but not to handle (in TED's RSS feed for example) -0500.

![image](https://cloud.githubusercontent.com/assets/2262919/12005365/bf8db252-ab66-11e5-82a0-ba422660785b.png)

**Comment by mitchdowney** [comment]
It appears that the NSDate+FeedsDate.swift from the FeedParser library we are using does not account for all possible DateFormats.

I implemented changes to the NSDate+FeedsDate.swift file based on [this SO answer](http://stackoverflow.com/a/19118926). Dan Carlin's podcasts are now correctly retrieving and saving dates.

I dunno yet if I broke another podcast's dates with this method...but all the podcasts I tested seem to be working with this new method. I pushed the fix to the branch fixEpisodeDates.
